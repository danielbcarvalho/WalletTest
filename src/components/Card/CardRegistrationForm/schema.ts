import * as yup from 'yup';
import { isValidCreditCard, isValidExpirationDate } from '../../../utils/card';

export const cardFormValidationSchema = yup.object({
  number: yup
    .string()
    .required('Número do cartão é obrigatório')
    .test('creditCard', 'Número do cartão inválido', value => {
      return value ? isValidCreditCard(value) : true;
    }),
  name: yup
    .string()
    .required('Nome do titular do cartão é obrigatório')
    .test('fullName', 'Nome incompleto', value => {
      return value ? value.trim().split(' ').length > 1 : true;
    }),
  expiry: yup
    .string()
    .required('Data de vencimento é obrigatória')
    .matches(/^\d{2}\/\d{2}$/, 'Data de vencimento inválida')
    .test('expirationDate', 'Data de vencimento inválida', value => {
      return value ? isValidExpirationDate(value) : true;
    }),
  cvv: yup
    .string()
    .required('Código de segurança é obrigatório')
    .matches(/^\d{3}$/, 'Código de segurança inválido'),
});
