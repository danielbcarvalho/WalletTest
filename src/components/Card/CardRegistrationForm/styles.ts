import styled from 'styled-components/native';

export const BottomInputWrapper = styled.TouchableOpacity`
  flex-direction: row;
  justify-content: space-between;
`;

export const ScrollView = styled.ScrollView.attrs({
  showsVerticalScrollIndicator: false,
  contentContainerStyle: { flexGrow: 1 },
})``;
