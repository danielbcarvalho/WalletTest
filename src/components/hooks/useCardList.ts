import { useState } from 'react';
import { AxiosError } from 'axios';
import { useNavigation } from '@react-navigation/native';
import { useQuery, useQueryClient } from '@tanstack/react-query';

import { Card } from '../../models/CardModels';
import { cardService } from '../../services/card';
import { QueryKeys } from '../../services/QueryKeys';
import { isCardOnTheTop, topCardIndex } from '../../utils/card';
import { CardListScreenProps } from '../../screens/CardListScreen';

const FIVE_MINUTES = 1000 * 60 * 5;

export function useCardList() {
  const queryClient = useQueryClient();
  const { navigate } = useNavigation<CardListScreenProps>();
  const [useThisCard, setUseThisCard] = useState<boolean>(false);

  const {
    error,
    isError,
    data: cards,
    isInitialLoading,
  } = useQuery<Card[]>([QueryKeys.CARD_LIST], () => cardService.list(), {
    staleTime: FIVE_MINUTES,
  });

  function setCardOnTop(selectedCardId: string) {
    if (cards) {
      const cardSelectedIndex = cards.findIndex(
        card => card.id === selectedCardId,
      );

      if (!isCardOnTheTop(cardSelectedIndex, cards)) {
        queryClient.setQueryData(
          [QueryKeys.CARD_LIST],
          [cards[topCardIndex(cards)], cards[cardSelectedIndex]],
        );
      }
    }
  }

  function onUseThisCard() {
    setUseThisCard(true);
    navigate('CardPayment');
  }

  return {
    cards: cards ?? [],
    error: error as AxiosError | undefined,
    isError,
    useThisCard,
    setCardOnTop,
    onUseThisCard,
    isLoading: isInitialLoading,
  };
}
