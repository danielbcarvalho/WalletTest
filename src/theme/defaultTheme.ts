import { DefaultTheme } from 'styled-components/native';

export const defaultTheme: DefaultTheme = {
  pallete: {
    primary: {
      main: '#142995',
    },
    success: {
      main: '#A5FF32',
    },
    alert: {
      main: '#FF0000',
    },
    warning: {
      main: '#FFB800',
    },
    info: {
      main: '#12C2E9',
    },
    background: {
      default: '#142995',
      overlay: '#EEEEEE',
    },
    textOnBackground: {
      default: '#FFF',
      overlay: '#142995',
    },
  },
  typography: {
    h1: {
      fontFamily: 'PTSans-Caption',
      fontSize: '28px',
      lineHeight: '42px',
    },
    h2: {
      fontFamily: 'PTSans-Caption',
      fontSize: '26px',
      lineHeight: '38px',
    },
    h3: {
      fontFamily: 'PTSansCaption',
      fontSize: '22px',
      lineHeight: '34px',
    },
    h4: {
      fontFamily: 'PTSans-Regular',
      fontSize: '20px',
      lineHeight: '32px',
    },
    h5: {
      fontFamily: 'PTSans-Regular',
      fontSize: '18px',
      lineHeight: '30px',
    },
    p: {
      fontFamily: 'PTSans-Regular',
      fontSize: '16px',
      lineHeight: '28px',
    },
    pSmall: {
      fontFamily: 'PTSans-Regular',
      fontSize: '14px',
      lineHeight: '26px',
    },
  },
  components: {
    button: {
      radius: '12px',
      fontFamily: 'PTSans-Regular',
      fontSize: '18px',
      lineHeight: '28px',
      primary: {
        backgroundColor: '#12C2E9',
        color: '#FFFFFF',
      },
      secondary: {
        backgroundColor: '#A5FF32',
        color: '#142995',
      },
      disabled: {
        backgroundColor: '#EEEEEE',
        color: '#BBBBBB',
      },
    },
    input: {
      label: {
        fontFamily: 'PTSans-Regular',
        fontSize: '14px',
        color: '#FFFFFF',
        lineHeight: '16px',
      },
      fontFamily: 'PTSans-Regular',
      fontSize: '16px',
      color: '#111',
      backgroundColor: '#EEEEEE',
      borderRadius: '12px',
      padding: '16px',
    },
    header: {
      title: {
        fontFamily: 'PTSans-Caption',
        fontSize: '22px',
        lineHeight: '24px',
        register: {
          color: '#142995',
          backgroundColor: '#FFFFFF',
        },
        cardList: {
          color: '#12C2E9',
        },
      },
    },
    card: {
      black: {
        backgroundColor: '#000000',
        color: '#FFFFFF',
      },
      green: {
        backgroundColor: '#A5FF32',
        color: '#3F3F3F',
      },
      title: {
        fontFamily: 'PTSans-Regular',
        fontSize: '18px',
        lineHeight: '20px',
      },
      name: {
        fontFamily: 'PTSans-Regular',
        fontSize: '16px',
        lineHeight: '18px',
      },
      text: {
        fontFamily: 'PTSans-Regular',
        fontSize: '14px',
        lineHeight: '16px',
      },
    },
  },
};

export const mockDefaultTheme: object = defaultTheme;
