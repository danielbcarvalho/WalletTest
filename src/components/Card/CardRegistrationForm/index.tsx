import React from 'react';

import { useTranslation } from 'react-i18next';
import { Masks } from 'react-native-mask-input';
import { yupResolver } from '@hookform/resolvers/yup';
import { useForm, Controller } from 'react-hook-form';
import { useFocusEffect } from '@react-navigation/native';

import { Button } from '../../Button';
import InputCustom from '../../InputCustom';

import { BottomInputWrapper } from './styles';
import { cardFormValidationSchema } from './schema';
import { getMaskCVV, getMaskDateMMYY } from '../../../utils/mask-input';

interface Props {
  onCardRegisterForm: (card: FormValues) => void;
}

interface FormValues {
  number: string;
  name: string;
  expiry: string;
  cvv: string;
}

function CardRegistrationForm({ onCardRegisterForm }: Props) {
  const { t } = useTranslation();
  const {
    control,
    handleSubmit,
    formState,
    reset: clearForm,
  } = useForm<FormValues>({
    defaultValues: {
      number: '',
      name: '',
      expiry: '',
      cvv: '',
    },
    resolver: yupResolver(cardFormValidationSchema),
    mode: 'all',
  });

  const onSubmit = (data: FormValues) => {
    const card: FormValues = {
      name: data.name,
      number: data.number,
      cvv: data.cvv,
      expiry: data.expiry,
    };
    onCardRegisterForm(card);
  };

  useFocusEffect(
    React.useCallback(() => {
      const unsubscribe = () => clearForm();

      return () => unsubscribe();
    }, [clearForm]),
  );

  return (
    <>
      <Controller
        control={control}
        render={({ field: { onChange, onBlur, value } }) => {
          return (
            <InputCustom
              accessibilityLabel={t('card number')}
              onBlur={onBlur}
              onChangeText={onChange}
              value={value}
              label={t('card number')}
              placeholder=""
              keyboardType="numeric"
              mask={Masks.CREDIT_CARD}
              name="number"
            />
          );
        }}
        name="number"
      />
      <Controller
        control={control}
        render={({ field: { onChange, onBlur, value } }) => (
          <InputCustom
            accessibilityLabel={t('card holder name')}
            onBlur={onBlur}
            onChangeText={onChange}
            value={value}
            label={t('card holder name')}
            keyboardType="default"
            autoCapitalize="words"
            name="name"
          />
        )}
        name="name"
      />
      <BottomInputWrapper>
        <Controller
          control={control}
          render={({ field: { onChange, onBlur, value } }) => (
            <InputCustom
              accessibilityLabel={t('expiration date')}
              placeholder="00/00"
              onBlur={onBlur}
              onChangeText={onChange}
              value={value}
              label={t('expiration date')}
              keyboardType="numeric"
              width={48}
              mask={getMaskDateMMYY}
              name="expiry"
            />
          )}
          name="expiry"
        />
        <Controller
          control={control}
          render={({ field: { onChange, onBlur, value } }) => (
            <InputCustom
              accessibilityLabel={t('cvv')}
              placeholder="***"
              onBlur={onBlur}
              onChangeText={onChange}
              value={value}
              label={t('cvv')}
              keyboardType="numeric"
              width={48}
              mask={getMaskCVV}
              name="cvv"
            />
          )}
          name="cvv"
        />
      </BottomInputWrapper>
      <Button
        title={t('advance')}
        accessibilityLabel={t('advance')}
        onPress={handleSubmit(onSubmit)}
        disabled={!formState.isValid}
      />
    </>
  );
}

export default CardRegistrationForm;
