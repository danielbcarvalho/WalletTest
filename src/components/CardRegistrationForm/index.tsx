import React from 'react';

import { Masks } from 'react-native-mask-input';
import { yupResolver } from '@hookform/resolvers/yup';
import { useForm, Controller } from 'react-hook-form';

import { Button } from '../Button';
import InputCustom from '../InputCustom';
import { CardProps } from '../../screens/CardRegistrationScreen';

import { BottomInputWrapper } from './styles';
import { cardFormValidationSchema } from './schema';
import { getMaskCVV, getMaskDateMMYY } from '../../utils/mask-input';
import { useTranslation } from 'react-i18next';

interface Props {
  onCardRegisterForm: (card: CardProps) => void;
}

interface FormValues {
  card: string;
  name: string;
  date: string;
  cvv: string;
}

function CardRegistrationForm({ onCardRegisterForm }: Props) {
  const { t } = useTranslation();
  const { control, handleSubmit, formState } = useForm<FormValues>({
    defaultValues: {
      card: '',
      name: '',
      date: '',
      cvv: '',
    },
    resolver: yupResolver(cardFormValidationSchema),
  });

  const onSubmit = (data: FormValues) => {
    const card: CardProps = {
      name: data.name,
      card: data.card,
      cvv: data.cvv,
      date: data.date,
    };
    onCardRegisterForm(card);
  };

  return (
    <>
      <Controller
        control={control}
        render={({ field: { onChange, onBlur, value } }) => (
          <InputCustom
            accessibilityLabel={t('card number')}
            onBlur={onBlur}
            onChangeText={onChange}
            value={value}
            label={t('card number')}
            placeholder=""
            keyboardType="numeric"
            mask={Masks.CREDIT_CARD}
          />
        )}
        name="card"
      />
      <Controller
        control={control}
        render={({ field: { onChange, onBlur, value } }) => (
          <InputCustom
            accessibilityLabel={t('card holder name')}
            onBlur={onBlur}
            onChangeText={onChange}
            value={value}
            label={t('card holder name')}
            keyboardType="default"
            autoCapitalize="words"
          />
        )}
        name="name"
      />
      <BottomInputWrapper>
        <Controller
          control={control}
          render={({ field: { onChange, onBlur, value } }) => (
            <InputCustom
              accessibilityLabel={t('expiration date')}
              placeholder="00/00"
              onBlur={onBlur}
              onChangeText={onChange}
              value={value}
              label={t('expiration date')}
              keyboardType="numeric"
              width={48}
              mask={getMaskDateMMYY}
            />
          )}
          name="date"
        />
        <Controller
          control={control}
          render={({ field: { onChange, onBlur, value } }) => (
            <InputCustom
              accessibilityLabel={t('cvv')}
              placeholder="***"
              onBlur={onBlur}
              onChangeText={onChange}
              value={value}
              label={t('cvv')}
              keyboardType="numeric"
              width={48}
              mask={getMaskCVV}
            />
          )}
          name="cvv"
        />
      </BottomInputWrapper>
      <Button
        title={t('advance')}
        accessibilityLabel={t('advance')}
        onPress={handleSubmit(onSubmit)}
        disabled={!formState.isValid}
      />
    </>
  );
}

export default CardRegistrationForm;
