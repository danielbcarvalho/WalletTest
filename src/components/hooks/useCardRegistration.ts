import uuid from 'react-native-uuid';
import { useTranslation } from 'react-i18next';
import { useNavigation } from '@react-navigation/native';
import { useMutation, useQueryClient } from '@tanstack/react-query';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import { ShowAlert } from '../ShowAlert';
import { ParamsList } from '../../Routes';
import { Card } from '../../models/CardModels';
import { cardService } from '../../services/card';
import { QueryKeys } from '../../services/QueryKeys';

import { useCardList } from './useCardList';

type CardRegistrationScreenProps = NativeStackNavigationProp<
  ParamsList,
  'CardRegistration'
>;

interface CardRegistrationFormData {
  cvv: string;
  name: string;
  number: string;
  expiry: string;
}

export function useCardRegistration() {
  const { t } = useTranslation();
  const navigation = useNavigation<CardRegistrationScreenProps>();
  const queryClient = useQueryClient();

  const { cards } = useCardList();

  const registerCardMutation = useMutation(cardService.register, {
    onSuccess: data => {
      queryClient.invalidateQueries([QueryKeys.CARD_LIST]);
      navigation.push('Home');
      navigation.navigate('CardRegistrationSuccess', { card: data });
    },
    onError: (error: any) => {
      ShowAlert({
        title: 'Error',
        message: error.message,
        onPress: () => navigation.navigate('Home'),
      });
    },
  });

  const handleCardRegisterForm = async (data: CardRegistrationFormData) => {
    try {
      if (cards.length >= 2) {
        ShowAlert({
          title: t('Ops, something went wrong'),
          message: t('Soon you will be able to register more cards'),
          onPress: () => navigation.navigate('Home'),
        });
      } else {
        const cardData: Card = {
          id: uuid.v4() as string,
          kind: 'green',
          ...data,
        };

        await registerCardMutation.mutateAsync(cardData);
      }
    } catch (error: any) {
      ShowAlert({
        title: t('Ops, something went wrong'),
        message: t('Try again later'),
        onPress: () => navigation.navigate('Home'),
      });
    }
  };

  return handleCardRegisterForm;
}
