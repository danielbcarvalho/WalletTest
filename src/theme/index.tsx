import React, { createContext, useMemo } from 'react';
import { ThemeProvider as ThemeProviderStyled } from 'styled-components/native';

import { defaultTheme } from './defaultTheme';

export enum Theme {
  light = 'light',
  dark = 'dark',
}

interface IApplication {
  children: JSX.Element;
}

const themes = {
  light: defaultTheme,
};

export const ThemeContext = createContext({
  theme: Theme.light,
});

export function ThemeProvider({ children }: IApplication) {
  const theme = Theme.light;

  const memoizedValue = useMemo(() => ({ theme }), [theme]);

  return (
    <ThemeContext.Provider value={memoizedValue}>
      <ThemeProviderStyled theme={themes[theme]}>
        {children}
      </ThemeProviderStyled>
    </ThemeContext.Provider>
  );
}
