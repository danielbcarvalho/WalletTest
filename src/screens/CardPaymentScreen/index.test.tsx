import React from 'react';
import { act } from 'react-test-renderer';
import * as ReactQuery from '@tanstack/react-query';
import { useNavigation } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import CardRegistrationForm from '../../components/Card/CardRegistrationForm';
import { useCardRegistration } from '../../components/hooks/useCardRegistration';
import CardRegistration from './';
import { renderHook } from '@testing-library/react-hooks';
import { useMutation, useQueryClient } from '@tanstack/react-query';
import { ThemeProvider } from '../../theme';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
  useFocusEffect: jest.fn(),
}));

jest.mock('@tanstack/react-query', () => ({
  useMutation: jest.fn(),
  useQueryClient: jest.fn(),
  useQuery: jest.fn(),
}));

describe('CardRegistrationScreen', () => {
  const mockNavigate = jest.fn();
  const mockInvalidateQueries = jest.fn();
  const mockMutateAsync = jest.fn();
  const mockData = {
    id: '4d18544b-66c4-478b-ab4a-0bab14c6f4d8',
    number: '5503 2328 4377 9802',
    name: 'John Doe',
    expiry: '12/24',
    kind: 'black',
    cvv: '123',
  };

  beforeEach(() => {
    jest.spyOn(ReactQuery, 'useQuery').mockImplementation(
      jest.fn().mockReturnValue({
        data: [],
      }),
    );

    useNavigation.mockReturnValue({ navigate: mockNavigate });
    useQueryClient.mockReturnValue({
      invalidateQueries: mockInvalidateQueries,
    });
    useMutation.mockReturnValue({
      mutateAsync: mockMutateAsync,
      onError: jest.fn(),
    });
    ReactQuery.useQuery.mockReturnValue({
      data: mockData,
      onError: jest.fn(),
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
    jest.clearAllMocks();
  });

  test('call useCardRegistration hook and trigger onCardRegisterForm on form submission', async () => {
    const onCardRegisterFormMock = jest.fn();

    const { result } = renderHook(() => useCardRegistration());

    await act(async () => {
      await result.current(mockData);
    });

    const { getByLabelText, getByText } = render(
      <ThemeProvider>
        <CardRegistrationForm onCardRegisterForm={onCardRegisterFormMock} />
      </ThemeProvider>,
    );
    await act(async () => {
      fireEvent.changeText(
        getByLabelText('card number'),
        '5284 7474 3246 1715',
      );
      fireEvent.changeText(getByLabelText('card holder name'), 'John Doe');
      fireEvent.changeText(getByLabelText('expiration date'), '12/23');
      fireEvent.changeText(getByLabelText('cvv'), '123');
    });
    await act(async () => {
      fireEvent.press(getByText('advance'));
    });

    await waitFor(() =>
      expect(onCardRegisterFormMock).toHaveBeenCalledTimes(1),
    );
  });

  test('matches snapshot', async () => {
    const { result } = renderHook(() => useCardRegistration());

    await act(async () => {
      await result.current(mockData);
    });

    const { toJSON } = render(
      <ThemeProvider>
        <CardRegistration />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  // test('matches snapshot', async () => {
  //   jest.spyOn(ReactQuery, 'useQuery').mockImplementation(
  //     jest.fn().mockReturnValue({
  //       data: mockData,
  //     }),
  //   );

  //   const { result } = renderHook(() => useCardRegistration());

  //   await act(async () => {
  //     await result.current(mockData);
  //   });

  //   const { toJSON } = render(
  //     <ThemeProvider>
  //       <CardRegistration />
  //     </ThemeProvider>,
  //   );
  //   expect(toJSON()).toMatchSnapshot();
  // });
});
