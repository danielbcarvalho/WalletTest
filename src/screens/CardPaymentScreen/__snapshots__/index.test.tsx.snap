// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRegistrationScreen matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 0,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      ListFooterComponent={[Function]}
      data={
        {
          "cvv": "123",
          "expiry": "12/24",
          "id": "4d18544b-66c4-478b-ab4a-0bab14c6f4d8",
          "kind": "black",
          "name": "John Doe",
          "number": "5503 2328 4377 9802",
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={{}}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginBottom": 20,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#12C2E9",
                  "borderBottomLeftRadius": 12,
                  "borderBottomRightRadius": 12,
                  "borderTopLeftRadius": 12,
                  "borderTopRightRadius": 12,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 10,
                  "maxHeight": 56,
                  "minHeight": 56,
                  "opacity": 1,
                  "paddingBottom": 10,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 10,
                }
              }
              testID="multipurpose-button"
            >
              <Text
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "PTSans-Regular",
                    "fontSize": 18,
                  }
                }
                type="primary"
              >
                pay with this card
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
